src/rulesets/oas3/index.json

```diff
@@ -83,7 +83,20 @@
       "message": "{{error}}",
       "recommended": true,
       "type": "validation",
-      "given": "$..[?(@.example)]",
+      "given": "$..[?(@.example && !@.type)]",
+      "then": {
+        "function": "schemaPath",
+        "functionOptions": {
+          "field": "example",
+          "schemaPath": "$.schema"
+        }
+      }
+    },
+    "valid-type-example": {
+      "summary": "Examples must be valid against their type.",
+      "message": "{{error}}",
+      "type": "validation",
+      "given": "$..[?(@.example && @.type)]",
```

Comment:
ditto

The problem with `type` is that I'm not sure there isn't a nasty corner case where that would fail.